### PR TITLE
5387- Upgrade images and pin flake8

### LINKED ISSUE #5387: Update deprecated Docker convenience images for circleci
### What we’re after

We should update our docker convenience images as the ones we are currently using have been deprecated. Our builds have been showing this warning:
Screen Shot 2022-07-21 at 7 50 27 PM

[Here's the discussion ](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034)about the new images. 

There's two instances in our code which need to be replaced, both in our config.yml file:
circleci/postgres:13.5→ [cimg/postgres:13.5](https://circleci.com/developer/images/image/cimg/postgres?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--uscan-en-dsa-maxConv-auth-brand&utm_term=g_-_c__dsa_&utm_content=&gclid=Cj0KCQjw8uOWBhDXARIsAOxKJ2HGV9wTP8B36ugjehyOg8J5xTIjgaIfehUpWxkGKEiuypa4bXg5F2MaArEvEALw_wcB)
circleci/python:3.8.12-buster → [cimg/python:3.8.12](https://circleci.com/developer/images/image/cimg/python)

### Action item(s)

(These are the smaller tasks that should happen in order to complete this work)
- [x] Update our images in fec-cms
- [x] Test the new images/make sure all of the necessary libraries are being installed

### Completion criteria

(What does the end state look like - as long as this task(s) is done, this work is complete)
- [x] We are no longer getting this warning or have confirmed that this warning can not be fixed currently.
- [ ] We have updated the images for the other repos or have built tickets for the other repos
       

    - [x]  OPENFEC
     - [x] PROXY
     - [x] EREGS
     - [ ] PATTERN LIB

### References/resources/technical considerations
NOTE: When migrating to a next-gen image, there might be some software issues. Common issues include:
A library you were using now has a different version. An apt package is no longer pre-installed. In this scenario simply install that package using:
`sudo apt-get update && sudo apt-get install -y <the-package>`

### PR DESCRIPTION
**Pull request by cnlucas**
## Summary (required)

- Resolves #5387

This updates the docker images and pins flake8 until we can fix our tests failing in the most recent major version. That ticket is [here. ](https://github.com/fecgov/openFEC/issues/5217)

### Required reviewers

1-2 devs

## Impacted areas of the application

General components of the application that this PR will affect:

-  circleci config

## Related PRs
https://github.com/fecgov/openFEC/issues/5217

## How to test

I tried to test run circleci config locally via their CLI tools and wasn't able to run them successfully. I got some errors that they seems to know about and aren't moving forward with right now. I did deploy to dev and everything ran successfully. 
-checkout this branch
-deploy to dev

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,19 +9,18 @@ jobs:
       # use `-browsers` prefix for selenium tests, e.g. `<image_name>-browsers`
 
       # Python
-      - image: circleci/python:3.8.12-buster-node-browsers
+      - image: cimg/python:3.8.12-browsers
```
Comment (patphongs) [inline .circleci/config.yml:12]: On our environment, I see a small patch version up from this version, can you try `python 3.8.13`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,7 +31,7 @@ jobs:
           # https://circleci.com/docs/2.0/postgres-config/#postgresql-circleci-configuration-example
           command: |
             sudo apt-get update -qq && sudo apt-get install -y build-essential postgresql-client
-            echo 'export PATH=/usr/lib/postgresql/10.11/bin/:$PATH' >> $BASH_ENV
+            echo 'export PATH=/usr/lib/postgresql/11.16/bin/:$PATH' >> $BASH_ENV
```
Comment (patphongs) [inline .circleci/config.yml:34]: If we update the postgresql version to 12.8, this will need to change as well

**Review by patphongs — COMMENTED** [review]
@cnlucas Thanks for these updates! I did a little digging to see our versions of python and postgresql being run in our environment and I think it may be good to make them the same versions on the circle image.

File: .circleci/config.yml [hunk 3]
```diff
@@ -9,19 +9,18 @@ jobs:
...
       # Python
-      - image: circleci/python:3.8.12-buster-node-browsers
+      - image: cimg/python:3.8.12-browsers
         environment:
           TZ: America/New_York
           DATABASE_URL: postgres://postgres@0.0.0.0/cfdm_cms_test
 
       # PostgreSQL
-      - image: circleci/postgres:11.9
+      - image: cimg/postgres:11.16
```
[the remark this answers is not in the document]
Comment (cnlucas) [reply .circleci/config.yml:18]: Sure thing, will update.

**Comment by cnlucas** [comment]
closing in favor of https://github.com/fecgov/fec-cms/pull/5395

**Pull request closed without merging**